src/mlpack/core/optimizers/adam/adam_impl.hpp

```diff
@@ -67,6 +70,9 @@ double Adam<DecomposableFunctionType>::Optimize(arma::mat& iterate)
   // Exponential moving average of gradient values.
   arma::mat m = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
 
+  // Initialize the exponentially weighted infinity norm for AdaMax optimizer.
+  arma::mat u = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
```

Comment:
Yes, I agree. Should we extend that approach to look something like the following piece of code?
```cpp
arma::mat u, v;
  if (adaMax)
    u = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
  else
    v = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
```
 I think it'd more intuitive if preceeded by a comment on what u and v are, and used whether for Adam or Adamax e.g. 

> Initialize, either the exponentially weighted infinity norm for AdaMax optimizer (u) or exponential moving average of squared gradient values for Adam optimizer (v).

What do you suggest should be done?